modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,159 @@
+          perform_ldap_query(ldap, filter, entries)
+
+        when 'ENUM_COMPUTERS'
+          filter = Net::LDAP::Filter.construct('(&(objectCategory=Computer)(userAccountControl:1.2.840.113556.1.4.803:=8192))') # Find domain controllers
+          perform_ldap_query(ldap, filter, entries)
+
+          filter = Net::LDAP::Filter.construct('(objectCategory=Computer)') # Find computers
+          perform_ldap_query(ldap, filter, entries)
+
+        when 'ENUM_EXCHANGE'
+          filter = Net::LDAP::Filter.construct('(&(objectClass=msExchExchangeServer)(!(objectClass=msExchExchangeServerPolicy)))') # Find Exchange Servers
+          perform_ldap_query(ldap, filter, entries)
+          filter = Net::LDAP::Filter.construct('(mailNickname=*)') # Find Exchange Recipients
+          perform_ldap_query(ldap, filter, entries)
+          # filter = Net::LDAP::Filter.construct("(&(msExchHideFromAddressLists=TRUE)(!objectClass=publicFolder))") # Find Exchange Recipients - hidden
+          # perform_ldap_query(ldap, filter, entries)
+          filter = Net::LDAP::Filter.construct('(proxyAddresses=FAX:*)') # Find Exchange Recipients - with FAX address
+          perform_ldap_query(ldap, filter, entries)
+
+        when 'ENUM_GROUPS'
+          filter = Net::LDAP::Filter.construct('(|(objectClass=group)(objectClass=groupOfNames))') # Standard LDAP groups query.
+          perform_ldap_query(ldap, filter, entries)
+
+          filter = Net::LDAP::Filter.construct('(groupType:1.2.840.113556.1.4.803:=2147483648)') # Find security groups within an AD environment.
+          perform_ldap_query(ldap, filter, entries)
+
+          filter = Net::LDAP::Filter.construct('(objectClass=posixGroup)') # Find Linux groups
+          perform_ldap_query(ldap, filter, entries)
+
+        when 'ENUM_ORGUNITS'
+          filter = Net::LDAP::Filter.construct('(objectClass=organizationalUnit)') # Find OUs aka Organizational Units
+          perform_ldap_query(ldap, filter, entries)
+
+        when 'ENUM_ORGROLES'
+          filter = Net::LDAP::Filter.construct('(objectClass=organizationalRole)') # Find OUs aka Organizational Units
+          perform_ldap_query(ldap, filter, entries)
+
+        when 'ENUM_PEOPLE'
+          filter = Net::LDAP::Filter.construct('(objectClass=organizationalPerson)') # Find people within an organization by Person entries.
+          perform_ldap_query(ldap, filter, entries)
+
+        when 'ENUM_USERS'
+          filter = Net::LDAP::Filter.construct('(|(objectClass=inetOrgPerson)(objectClass=user))') # Common LDAP user query.
+          perform_ldap_query(ldap, filter, entries)
+
+          filter = Net::LDAP::Filter.construct('(sAMAccountType=805306368)') # Common AD User query by account type.
+          perform_ldap_query(ldap, filter, entries)
+
+          filter = Net::LDAP::Filter.construct('(objectClass=posixAccount)') # Query for Linux accounts.
+          perform_ldap_query(ldap, filter, entries)
```

Comment:
I'm not super familiar with ldap, but am I right in saying this can be performed in a single query? 👀

Reply:
Umm I think it should be possible, fair point.

Reply:
`(|(objectClass=inetOrgPerson)(objectClass=user)(objectClass=posixAccount)(sAMAccountType=805306368))` Should solve this but I'd be hesitant to make the query too long, not sure if there is a limit or not.

Reply:
@adfoster-r7 So one question I did have here is whether or not we really want to do this vs parsing the queries separately as each one is likely to contain different data. 

Though I suppose if we are just dumping data that might not be so bad, but my fear is more that we combine data of different types into the dump making it harder to parse.